swing/src/net/sf/openrocket/gui/main/SimulationPanel.java

```diff
@@ -598,6 +624,216 @@ public void updateEnabledState() {
 		}
 	}
 
+	class DumpSimulationToCSVAction extends SimulationAction {
+		private String lastSelectedLocation = "";
+		private HashMap<String, String> cToUnit;
+		public DumpSimulationToCSVAction() {
+			putValue(NAME, trans.get("simpanel.pop.export_to_csv"));
+			putValue(SMALL_ICON, Icons.FILE_EXPORT_AS);
+		}
+
+		/**
+		 * Means by wich the CSV export will clean up units on the values and
+		 * describe them on the header fields instead.
+		 */
+		private void populateColumnNameToUnitsHashTable() {
+			if (null == simulationTableModel) {
+				return;
+			}
+			valueColumnToUnitString.clear(); // necessary if units changed during session
+			for (int i=0; i<simulationTableModel.getColumnCount(); i++) {
+				Column c = simulationTableModel.getColumn(i);
+				if (c instanceof ValueColumn) {
+					// only value columns seem to have units that are not zero length strings... these are
+					// the ones we actually want in our lookup table.
+					valueColumnToUnitString.put(c.toString(), c.getUnits().getDefaultUnit().getUnit());
+				}
+				
+			}
+		}
+		/**
+		 * Dump data from sim table to file for run simulations
+		 * @param data The csv data as one string block.
+		 * @param csvFile The file to dump the data to.
+		 */
+		private void dumpDataToFile(String data, File csvFile) {
+			BufferedWriter bufferedWriter = null;
+			try {
+				csvFile.createNewFile();
+				bufferedWriter = new BufferedWriter(new FileWriter(csvFile));
+				bufferedWriter.write(data);
+				this.lastSelectedLocation = csvFile.getParent();
+			} catch (FileNotFoundException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} catch (IOException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} finally {
+				if (bufferedWriter != null) {
+					try {
+						bufferedWriter.close();
+					} catch (IOException e) {
+						String msg = e.getMessage();
+						JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+					}
+				}
+			}
+		}
+
+		private JFileChooser setUpFileChooser() {
+			JFileChooser fch = new JFileChooser();
```

Comment:
I would add a `CsvOptionPanel` so you can choose the CSV parameters in the file chooser dialog. See the example in `swing/src/net/sf/openrocket/gui/dialogs/optimization/GeneralOptimizationDialog.java savePath()`

Reply:
Each enough to parrot... but it adds more than the separation char, and it's unclear if the decimal precision can easily be applied to the strings extracted from the table...

Reply:
Well you do the String conversion yourself in the code:
```Java
String value = o.toString();
```
You can try this instead:
```Java
final String valueString;
if (o instanceof Value) {
    double value = ((Value) o).getValue();
    valueString = {implement the code to format the double using the supplied decimal precision};
} else {
    valueString = o.toString();
}
```